extensions/mssql/src/objectExplorerNodeProvider/objectExplorerNodeProvider.ts

```diff
@@ -0,0 +1,348 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import * as sqlops from 'sqlops';
+import * as vscode from 'vscode';
+import * as nls from 'vscode-nls';
+const localize = nls.loadMessageBundle();
+
+import * as UUID from 'vscode-languageclient/lib/utils/uuid';
+import { ProviderBase } from './providerBase';
+import { Connection } from './connection';
+import * as utils from '../utils';
+import { TreeNode } from './treeNodes';
+import { ConnectionNode, TreeDataContext, ITreeChangeHandler } from './hdfsProvider';
+import { IFileSource } from './fileSources';
+import { AppContext } from '../appContext';
+import * as constants from '../constants';
+
+const outputChannel = vscode.window.createOutputChannel(constants.providerId);
```

Comment:
Will this cause any duplicate channels? Just wondering if we need a shared one for the extension or if multiple classes can create this. Can you check with the core team / see how this works? I'm just worried about 2 MSSQL channels being opened by us.